### PR TITLE
Order added with Dao and Service layers

### PR DESCRIPTION
**Pull request by SetMyDream**

### CODE REVIEW

**Review by KostiantynVorobiov — APPROVED** [review]
ОК

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -64,38 +51,46 @@ public static void main(String[] args) throws AuthenticationException {
...
         ShoppingCartService cartService = (ShoppingCartService) injector
                 .getInstance(ShoppingCartService.class);
+        System.out.println(cartService.getByUser(johny));
+
         cartService.addSession(fastSession, den);
         cartService.addSession(groundHogDayManSession, johny);
-        System.out.println("Den's shopping cart :" + cartService.getByUser(den).toString());
+
         System.out.println("Johny's shopping cart :" + cartService.getByUser(johny).toString());
+        System.out.println("Den's shopping cart :" + cartService.getByUser(den).toString());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]:
```suggestion
        System.out.println("Johny's shopping cart :" + cartService.getByUser(johny));
        System.out.println("Den's shopping cart :" + cartService.getByUser(den));
```
calling `toString()` explicitly is redundant

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -64,38 +51,46 @@ public static void main(String[] args) throws AuthenticationException {
...
         ShoppingCart johnyShoppingCart = cartService.getByUser(johny);
         cartService.clear(johnyShoppingCart);
-        System.out.println("Roman's shoppingCart :" + cartService.getByUser(johny));
+        System.out.println("Johny's shoppingCart :" + cartService.getByUser(johny));
+
+        OrderService orderService = (OrderService) injector
+                .getInstance(OrderService.class);
+        orderService.completeOrder(johnyShoppingCart.getTickets(), johnyShoppingCart.getUser());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: you have just cleared `johnyShoppingCart`, is there any tickets there?

File: src/main/java/com/dev/cinema/models/Order.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        this.orderDate = orderDate;
+    }
+
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
+        this.user = user;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/models/Order.java]: `@Data` creates getters and setters for you

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -40,6 +40,7 @@ public void registerNewShoppingCart(User user) {
         ShoppingCart shoppingCart = new ShoppingCart();
         shoppingCart.setUser(user);
         shoppingCartDao.add(shoppingCart);
+        System.out.println(shoppingCart);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: do we need this?

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 5]
```diff
@@ -6,28 +6,24 @@
...
-                    = criteriaBuilder.createQuery(ShoppingCart.class);
-            Root<ShoppingCart> root = cartCriteriaQuery.from(ShoppingCart.class);
-            Predicate userIdPredicate = criteriaBuilder.equal(root.get("user"), user);
-            cartCriteriaQuery.select(root).where(userIdPredicate);
-            return session.createQuery(cartCriteriaQuery).uniqueResult();
+            Query<ShoppingCart> query = session.createQuery("FROM ShoppingCart sc"
+                    + " join fetch sc.user"
+                    + " left join fetch sc.tickets"
+                    + " WHERE sc.user.id = :id", ShoppingCart.class);
+            ((Query<?>) query).setParameter("id", user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java]: can we do without casting here?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/Main.java (hunk 2 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/Main.java]: no, 
but now den has

File: src/main/java/com/dev/cinema/models/Order.java (hunk 3 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/models/Order.java]: unfortunatly, not

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: no, removed)

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 5 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java]: yes

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 6]
```diff
@@ -19,7 +19,7 @@ public ShoppingCart getByUser(User user) {
                     + " join fetch sc.user"
                     + " left join fetch sc.tickets"
                     + " WHERE sc.user.id = :id", ShoppingCart.class);
-            ((Query<?>) query).setParameter("id", user.getId());
+            (query).setParameter("id", user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java]: why is `query` in parenthesis?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
let's also try to fix the lombok issue

**Comment by SetMyDream** [comment]
> let's also try to fix the lombok issue

now will try to deal with

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**